app/components/SpecialDay/AprilFools/LoadingBox.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+import React from 'react';
+import moment from 'moment-timezone';
+import _ from 'lodash';
+import styles from './LoadingBox.css';
+import walkingImage from 'app/assets/man_walking.gif';
+
+type Props = {
+  onQueueDone: () => void
+};
+
+type State = {
+  progress: number,
+  usersInQueue: number,
+  initialDate: moment,
+  initialTimeLeft: number,
+  expectedArrivalDate: moment,
+  secondsLeft: number,
+  lastUpdateTime: string,
+  queueId: string
+};
+
+class LoadingBox extends React.Component<Props, State> {
+  interval: IntervalID;
+
+  constructor(props: Props) {
+    super(props);
+    const initialDate = moment();
+
+    // Queue time between 20 and 57 seconds.
+    const expectedArrivalDate = moment().add(_.random(20, 57), 'seconds');
+    const initialTimeLeft = expectedArrivalDate.diff(initialDate);
+
+    this.state = {
+      progress: 0,
+      usersInQueue: _.random(48, 231), // between 48 and 231
+      initialDate,
+      initialTimeLeft,
+      expectedArrivalDate,
+      secondsLeft: Math.floor(initialTimeLeft / 1000),
+      lastUpdateTime: initialDate.format('HH:mm:ss'),
+      queueId: LoadingBox.randomString()
+    };
+  }
+
+  static randomString() {
+    return Math.random()
+      .toString(36)
+      .substring(2, 10);
+  }
+
+  updateProgress() {
+    const { onQueueDone } = this.props;
+    const { secondsLeft, initialTimeLeft, expectedArrivalDate } = this.state;
+    let { usersInQueue } = this.state;
+
+    const now = moment();
+    const timeLeft = expectedArrivalDate.diff(now);
+    let newSecondsLeft = Math.round(timeLeft / 1000);
+    let progress = 100 - (100 / initialTimeLeft) * timeLeft;
+
+    if (progress >= 100) {
+      clearInterval(this.interval);
+      onQueueDone();
+      return;
+    }
+
+    // Change users in queue every 5 seconds if usersInQueue is higher than 15.
+    if (
+      secondsLeft !== newSecondsLeft &&
+      secondsLeft % 5 === 0 &&
+      usersInQueue > 15
+    ) {
+      // Between 10 and 25
+      const randomCount = _.random(10, 25);
+      if (usersInQueue > 300) {
+        // Max 300 users.
+        usersInQueue -= randomCount;
+      } else if (usersInQueue < 50) {
+        // Min 30 users.
+        usersInQueue += randomCount;
+      } else {
+        // Random subtract/add count.
+        usersInQueue += Math.random() >= 0.5 ? randomCount : -randomCount;
+      }
+    }
+
+    this.setState({
+      progress,
+      usersInQueue,
+      secondsLeft: newSecondsLeft,
+      lastUpdateTime: now.format('HH:mm:ss')
+    });
+  }
+
+  componentDidMount() {
+    this.updateProgress();
+    this.interval = setInterval(() => this.updateProgress(), 50);
```

Comment:
Why so often (short interval)? Is it not enough to do this every second?

Reply:
It's more _smooth_.